File: use-cases/food-packaging/src/main/java/org/acme/common/ConstraintIdSanitizer.java
package org.acme.common;

import java.util.Objects;
import java.util.regex.Pattern;

public final class ConstraintIdSanitizer {

    private static final Pattern DISALLOWED_CHARACTERS = Pattern.compile("[^A-Za-z0-9 _'().-]");

    private ConstraintIdSanitizer() {
    }

    public static String sanitize(String constraintId) {
        Objects.requireNonNull(constraintId, "constraintId must not be null");
        return DISALLOWED_CHARACTERS.matcher(constraintId).replaceAll("-");
    }
}

File: getting-started/quarkus-integration-kotlin/src/main/kotlin/org/acme/common/ConstraintIdSanitizer.kt
package org.acme.common

object ConstraintIdSanitizer {

    private val disallowedCharacters = Regex("[^A-Za-z0-9 _'().-]")

    fun sanitize(constraintId: String): String = constraintId.replace(disallowedCharacters, "-")
}
